external-crates/move/crates/move-compiler/src/naming/translate.rs

```diff
@@ -2408,31 +2484,78 @@ fn lvalue(
                 C::Bind => "deconstructing binding",
                 C::Assign => "deconstructing assignment",
             };
-            let (m, sn, tys_opt, is_positional) =
+            let (m, sn, tys_opt, field_info) =
                 context.resolve_struct_name(loc, msg, tn, etys_opt)?;
-            if is_positional && !matches!(efields, E::FieldBindings::Positional(_)) {
+            if field_info.is_positional() && !matches!(efields, E::FieldBindings::Positional(_)) {
                 let msg = "Invalid deconstruction. Positional struct field declarations require \
                            positional deconstruction";
                 context
                     .env
                     .add_diag(diag!(NameResolution::PositionalCallMismatch, (loc, msg)));
             }
 
-            if !is_positional && matches!(efields, E::FieldBindings::Positional(_)) {
+            if !field_info.is_positional() && matches!(efields, E::FieldBindings::Positional(_)) {
                 let msg = "Invalid deconstruction. Named struct field declarations require \
                            named deconstruction";
                 context
                     .env
                     .add_diag(diag!(NameResolution::PositionalCallMismatch, (loc, msg)));
             }
+            let make_ignore = |loc| {
+                let var = sp(loc, Symbol::from("_"));
+                let name = E::ModuleAccess::new(loc, E::ModuleAccess_::Name(var));
+                sp(loc, E::LValue_::Var(None, name, None))
+            };
             let efields = match efields {
-                E::FieldBindings::Named(efields) => efields,
+                E::FieldBindings::Named(mut efields, ellipsis) => {
+                    if let Some(ellipsis_loc) = ellipsis {
+                        let mut fields = match field_info {
+                            FieldInfo::Empty => BTreeSet::new(),
+                            FieldInfo::Named(fields) => fields,
+                            FieldInfo::Positional(num_fields) => (0..num_fields)
+                                .map(|i| Field::add_loc(ellipsis_loc, format!("{i}").into()))
+                                .collect(),
+                        };
+
+                        for (k, _) in efields.key_cloned_iter() {
+                            fields.remove(&k);
+                        }
+
+                        let start_idx = efields.len();
+                        for (i, f) in fields.into_iter().enumerate() {
+                            efields
+                                .add(
+                                    Field(sp(ellipsis_loc, f.value())),
+                                    (start_idx + i, make_ignore(ellipsis_loc)),
+                                )
+                                .unwrap();
+                        }
+                    }
```

Comment:
Given that this code is in two places, I might like it as a helper function that is `Expansion -> Expansion` that we can use for both cases.